conjure-generator-common/src/main/java/com/palantir/conjure/CaseConverter.java

```diff
@@ -20,14 +20,16 @@
 import java.util.regex.Pattern;
 
 public final class CaseConverter {
-    public static final Pattern CAMEL_CASE_PATTERN = Pattern.compile("^[a-z]([A-Z]{1,2}[a-z0-9]|[a-z0-9])*[A-Z]?$");
-    public static final Pattern KEBAB_CASE_PATTERN =
-            Pattern.compile("^[a-z]((-[a-z]){1,2}[a-z0-9]|[a-z0-9])*(-[a-z])?$");
-    public static final Pattern SNAKE_CASE_PATTERN =
-            Pattern.compile("^[a-z]((_[a-z]){1,2}[a-z0-9]|[a-z0-9])*(_[a-z])?$");
+    public static final CachedPatternMatcher CAMEL_CASE_PATTERN =
+            CachedPatternMatcher.wrap(Pattern.compile("^[a-z]([A-Z]{1," + "2}[a-z0-9]|[a-z0-9])*[A-Z]?$"));
+    public static final CachedPatternMatcher KEBAB_CASE_PATTERN =
+            CachedPatternMatcher.wrap(Pattern.compile("^[a-z]((-[a-z]){1,2}[a-z0-9]|[a-z0-9])*(-[a-z])?$"));
+    public static final CachedPatternMatcher SNAKE_CASE_PATTERN =
+            CachedPatternMatcher.wrap(Pattern.compile("^[a-z]((_[a-z]){1,2}[a-z0-9]|[a-z0-9])*(_[a-z])?$"));
 
     private CaseConverter() {}
 
+    @SuppressWarnings("ImmutableEnumChecker")
```

Comment:
Up for discussion; the `CachedPatternMatcher` is technically not immutable, so the enum validator is correctly shouting, but it functionally acts like an immutable interface.